lib/string_reverse.rb

```diff
@@ -1,6 +1,15 @@
+# require "pry"
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: constant
 def string_reverse(my_string)
-  raise NotImplementedError
+  return "" if my_string == nil || my_string.length < 1
+  p = my_string.length / 2
+  q = my_string.length
+  p.times do |x|
+    temp = my_string[x]
+    my_string[x] = my_string[q - x - 1]

```

Comment:
Nit: `q - x - 1` is probably best moved into its own variable for two reasons:
* it's used multiple times
* it's not immediately clear what this represents - maybe `swap_index`?